### PR TITLE
[MWPW-147001] Stage initiative automation (Stage-Main sync PRs)

### PR DESCRIPTION
**Pull request by mokimo**
Based on the [discussion](https://github.com/orgs/adobecom/discussions/2206), the first workflow that will merge PRs into stage and open stage->main syncs.

PRs will be merged right after applying the `Ready for Stage` label if possible, or in a 24h interval. Once we have the stage->main MERGE workflow, that will also trigger opening a new stage->main sync instantly.

Resolves: [MWPW-147001](https://jira.corp.adobe.com/browse/MWPW-147001)

**Test URLs:**
- Before: https://main--milo--mokimo.hlx.live/?martech=off
- After: https://stage--milo--mokimo.hlx.live/?martech=off

### CODE REVIEW

File: .github/workflows/helpers.js
Comment (mokimo) [reply .github/workflows/helpers.js:26]: Yup, we still want in-action logs in the Github action, Slack is fast-paced feedback - while Github will be useful to debug the action when it ran it completeness
Screenshot 2024-04-30 at 17 28 12

File: .github/workflows/merge-to-stage.js
Comment (mokimo) [reply .github/workflows/merge-to-stage.js:10]: We don't require this at the moment either - 2 approvals & that counts. Code owners are just automatically looped in/requested for reviews

File: .github/workflows/merge-to-stage.js
Comment (mokimo) [reply .github/workflows/merge-to-stage.js:129]: Yup, also needed! It's very confusing to understand what happens, if the action just runs without any clear feedback

File: .github/workflows/merge-to-stage.js
Comment (mokimo) [reply .github/workflows/merge-to-stage.js:216]: We didn't really follow that rule to begin with 😕 

We mentioned we want that to clearly pin-point an offending PR + easy reverts and make it happen by checking that there is no file overlap.

So yes, we allow an arbitary number of PRs per batch, as long as they only touch [unique files](https://github.com/adobecom/milo/pull/2224/files#diff-ed9219c32bdd6c09304ebc81fc40de2930347168af4cb5e03f590c4375925bedR131)

File: .github/workflows/merge-to-stage.js
Comment (mokimo) [reply .github/workflows/merge-to-stage.js:47]: Hmm, maybe just a calendar entry for milo core?

File: .github/workflows/merge-to-stage.js
Comment (mokimo) [reply .github/workflows/merge-to-stage.js:129]: A bit of context while I'm at it: 
It's written in a way to be debug-able locally with a personal access token. The token is a personal access token is restricted to opening PRs, it can't merge anything.
So debugging is a mix of local + pushing the action to your fork and testing it on Github

**Review by 3ch023 — APPROVED** [review]

File: .github/workflows/merge-to-stage.js
Comment (3ch023) [reply .github/workflows/merge-to-stage.js:10]: true, but it's a good point from @overmyheadandbody , maybe in future we will need this..

File: .github/workflows/merge-to-stage.js
Comment (3ch023) [reply .github/workflows/merge-to-stage.js:47]: @mokimo  @overmyheadandbody  we could add a check if the current year is not 2024 anymore -> stop automerging completely.
this will ensure we will not forget to update a year :)

File: .github/workflows/merge-to-stage.js
Comment (3ch023) [reply .github/workflows/merge-to-stage.js:216]: I would probably still set some top number there.. like 10. or 7.
I know its unlikely, but very theoretically it could happen that SOTs won't approve for some time and people will keep and keep adding the PRs.

also this makes me think about a next issue.
right now we first 'form' the batch and only then invite SOTs to verify?
is it going to be the same with automation?

**Review by 3ch023 — COMMENTED** [review]
_got a question here 
https://github.com/adobecom/milo/pull/2224/files#r1587746577_

resolved

File: .github/workflows/merge-to-stage.js
Comment (mokimo) [reply .github/workflows/merge-to-stage.js:10]: Github offers this OOTB for a whole repo, we can still decide to enable this at some point
Screenshot 2024-05-02 at 16 38 28

So far it hasn't been an issue and hope it won't be in the future either - but the option is there

File: .github/workflows/merge-to-stage.js
Comment (mokimo) [reply .github/workflows/merge-to-stage.js:216]: If the first SOT label is applied, we don't merge anything into stage anymore. If (no) SOTs are not approving for a long time, we'd have to look into that bottleneck and re-plan accordingly 🤔 

> right now we first 'form' the batch and only then invite SOTs to verify?
is it going to be the same with automation?

Yes, unless you got better ideas, such as pre-stage?! which would bring us back to - just test on the branches 😁 it will be the same.

In theory, if we also have the stage->main MERGE workflow should get a little faster, it would be like this:

3/5 SOT approvals...
last SOT comes in and approves, 4/5 approvals there, PR is merged.
15 Seconds later you got your new batch and stage->main PR & the same SOT can instantly start testing and the new batch already has 1/5 approvals.

Comment (3ch023) [reply .github/workflows/merge-to-stage.js:216]: >If the first SOT label is applied, we don't merge anything into stage anymore. 

just realized that this will work good.
no matter how many PRs were in batch, since 1st SOT checked the batch and approved the rest will be testing the same code. all good

File: .github/workflows/merge-to-stage.js
Comment (overmyheadandbody) [reply .github/workflows/merge-to-stage.js:10]: Hmm, maybe someone de-activated it recently? A couple of months ago at least one code owner had to review the PR, even though it had more than 2 approvals already. And I'd say it's fair enough, that's one of the reasons we have the codeowners in the first place - automatically assign the specialized person and have them give their two cents. This might deserve some further discussion to understand who and why disabled this rule.

File: .github/workflows/merge-to-stage.js
Comment (mokimo) [reply .github/workflows/merge-to-stage.js:48]: Done

**Pull request merged**
